src/components/SelectEpisodes.jsx

```diff
@@ -1,26 +1,30 @@
-import React,{useState} from "react";
+import React,{useEffect, useState} from "react";
 
-const SelectEpisodes= ({isVisible,showId})=>{
+const SelectEpisodes= ({showId})=>{
 
   const[episodeData,setEpisodeData]=useState([]);
 
-   fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
-   .then(response=>response.json())
-   .then(data=>setEpisodeData(data))
-   .catch(err=>console.log(err));
+  useEffect(() => {
```

Comment:
I added this `useEffect` and put the `fetch` inside that. Although your code was working for the first click because of how React works, it was not the correct way of doing it.
If you have an `async` function to call, you need to be careful about it. Since `fetch` is `async`, in your case, it should be in a useEffect which is dependant to a `showId` (otherwise when `showId` changes by clicking on another show, the fetch doesn't get called)

Reply:
For fetching data I need to use hooks in each components that I use it. because there is a problem by clicking on each show card to filter it's episodes. maybe by using useMemo I can manage the data .

Reply:
> I added this `useEffect` and put the `fetch` inside that. Although your code was working for the first click because of how React works, it was not the correct way of doing it. If you have an `async` function to call, you need to be careful about it. Since `fetch` is `async`, in your case, it should be in a useEffect which is dependant to a `showId` (otherwise when `showId` changes by clicking on another show, the fetch doesn't get called)

thank you for your comment when we use
useEffect it gives us an error and we have not found out what is exactly is .

Reply:
@leilafaez I would suggest to make things work without `useMemo` first. When you are happy with the functionality, then refactor your code and introduce `useMemo`.

@baharbazargan62 can you tell me where do you see the error? (on my suggested code, or on your own code?) which steps do I need to take to reproduce the error?